vllm_spyre/worker/spyre_worker.py

```diff
@@ -35,6 +36,20 @@
 # yapf: enable
 
 
+# Capture a oneshot metric (e.x warmup)
+def _spyre_metric_oneshot(name, labels, value):
+    name = "vllm:plugin:spyre:" + name
+    # Cleanup existing, for CI.
+    if name in prometheus_client.REGISTRY._names_to_collectors:
+        collector = prometheus_client.REGISTRY._names_to_collectors[name]
+        prometheus_client.REGISTRY.unregister(collector)
+    gauge = prometheus_client.Gauge(name=name,
+                                    documentation="Oneshot spyre metric",
+                                    multiprocess_mode='mostrecent',
```

Comment:
v0 uses multiprocess mode to collect metrics from processes that aren't the main server process, but my last understanding of the prometheus integration in v1 was that it didn't use multiprocessing. I thought it instead collected all the stats to be reported as part of the `EngineCoreOutputs` class and then reported those from the frontend process.

If that's still true, then we're gonna have to do something a lot different to capture any extra metrics for v1

Reply:
Using the default multiprocess mode adds a pid label to each gauge. Instead, rank is used to differentiate each sampling. It can be removed without harm if we don't mind the extra noise.

Reply:
Right, what I'm saying is that I know this works for v0 because multiprocessing mode is enabled, but I think that trying this same strategy for v1 might just fail because vllm doesn't set up prometheus multiprocessing at all. So your `TODO: also port to V1 plugin` might require a very different design

Reply:
Looking into the V1 changes, I don't think we can use prometheus unless the multiprocess support is turned back on. I am not sure if or when that would happen given it is turned off for performance reasons.

I am not sure what other options beyond finding and scraping logs exist when using V1. Any hints?

Reply:
Dang, yeah that's what I suspected would be the case. I also doubt the multiprocess support will be turned back on, though we could ask.

We could write out the info to files? Then you could mount a shared remote storage when you deploy and run, and collect the data you need from there.